tools/data-handler/src/rename.ts

```diff
@@ -126,13 +151,37 @@ export class Rename extends EventEmitter {
       const copyLinkType = link.linkType.slice(0);
       link.cardKey = link.cardKey.replace(re, this.to);
       link.linkType = link.linkType.replace(re, this.to);
-      changed = copyCardKey !== link.cardKey || copyLinkType !== link.linkType;
+      changed =
+        changed ||
+        copyCardKey !== link.cardKey ||
+        copyLinkType !== link.linkType;
     });
     if (card.metadata && changed) {
       this.project.updateCardMetadata(card, card.metadata, true);
     }
   }
 
+  // Update card's macros
+  private async updateCardMacros(card: Card) {
```

Comment:
I think content! is a bit ugly. This function only depends on the content, so maybe just pass it?

Reply:
Yes, the method could just return if there is no content